run_tests.py

```diff
@@ -0,0 +1,284 @@
+
+'''
+
+from pathlib import Path, PurePath
+import subprocess
+import sys
+import getopt
+import os
+import json
+import time
+import re
+
+terminate_sketch_on_completion = False
+try:
+    import psutil
+    terminate_sketch_on_completion = True
+except ImportError:
+    print('Sketch will remain open after running the tests and must be terminated manually.', file=sys.stderr)
+
+def group_results_by_parent(results):
+    grouped_results = {}
+
+    for result in results:
+        parent_title = result['ancestorTitles'][0]  # test suite name
+        grouped_results[parent_title] = grouped_results.get(parent_title, {
+            'relativePath': result['relativePath'],
+            'results': [],
+        })
+
+        # drop the suite name from the ancestors
+        result['ancestorTitles'] = result['ancestorTitles'][1:]
+        grouped_results[parent_title]['results'].append(result)
+
+    return grouped_results
+
+
+def has_failed_tests(results):
+    for result in results:
+        if result['status'] == "failed":
+            return True
+
+    return False
+
+
+# Read 'com.apple.progress.fractionCompleted' extended file attribute from
+# ouput file to display the test runner progress.
+def watch_test_runner_progress(file_path, timeout=30):
+    progress = 0
+    last_progress_time = time.time()
+
+    print("Waiting for file to be created…")
+    # Wait for output file to be available on disk. This file is written
+    # at the beginning of the integration tests plugin run.
+    while not os.path.exists(file_path):
+        time.sleep(2)
+
+        if (time.time() - last_progress_time > timeout):
+            raise Exception("Timeout")
+
+    if not os.path.isfile(file_path):
+        raise Exception("Path to test output is not a file")
+
+    print("Waiting for tests to complete…")
+    while progress < 1:
+        stream = subprocess.Popen(
+            ['xattr', '-l', file_path],
+            stdout=subprocess.PIPE)
+
+        output = str(stream.communicate())
+        output = output.split("com.apple.progress.fractionCompleted:")[1]
+        match = re.search('\d+(.\d+)*', output)
+
+        latest_progress = float(match.group(0))
+        if latest_progress == progress:
+            # no progress since last run
+            if (time.time() - last_progress_time > timeout):
+                raise Exception("Timeout")
+
+            time.sleep(1)
+            continue
+
+        progress = latest_progress
+        print(f"Running tests: {round(progress * 100, 2)}% complete", end='\r')
+
+        time.sleep(1)
+        last_progress_time = time.time()
+
+
+def print_results(results):
+    
+    for suite_name, suite_results in results.items():
+
+        suite_status = "failed" if has_failed_tests(suite_results['results']) else "passed"
+
+        print('\n:: {status}\t{name} {relativePath}'.format(
```

Comment:
Any reason why you don't use an "f" formatted string? No need for .format() anymore.